### PR TITLE
Remove jQuery (part 1)

### PR DESCRIPTION
**Pull request by rubenarakelyan**
JIRA ticket: https://regulatorydelivery.atlassian.net/browse/PSD-1368

## Description

* Removes jQuery dependency from all JavaScript files at the top level of `app/javascript`
* Removes unused JavaScript
* Updates some syntax

### CODE REVIEW

File: app/views/help/about.html.erb [hunk 1]
```diff
@@ -470,8 +470,6 @@
   </div>
 </div>
 
-<%= render "close_page_button" %>
```
Comment (rubenarakelyan) [inline app/views/help/about.html.erb:473]: These buttons are never shown as far as I can see

File: app/views/investigations/creation_flow/about_enquiry.html.erb [hunk 2]
```diff
@@ -12,8 +12,8 @@
       <fieldset class="govuk-fieldset">
         <%= form.govuk_date_input :date_received, legend: "When was it received?" %>
         <p class="govuk-body">
-          Set to: <%= link_to('Today', 'javascript:void(0)', id: :today) %>
```
Comment (rubenarakelyan) [inline app/views/investigations/creation_flow/about_enquiry.html.erb:15]: We should not link to `javascript` links like this - accepted practice is to use `#` instead.

**Review by stdstaff — APPROVED** [review]
LGTM from the code perspective, please ask QA to test these components across [the standard browser matrix](https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices), and that the accessibility features are all still functioning as expected.

**Review by acallaghan — APPROVED** [review]

**Review by nasirkhanbeis — APPROVED** [review]
Regression tested this on the review app, looking good as expected.

**Pull request merged**